GoMeet/src/main/java/com/google/sps/servlets/VoteMeetingTimeServlet.java

```diff
@@ -0,0 +1,107 @@
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.PreparedQuery.TooManyResultsException;
+import com.google.sps.data.ErrorMessages;
+import com.google.sps.data.MeetingTimeFields;
+import com.google.sps.data.ServletUtil;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to handle the fetching and creating of MeetingTime entities */
+@WebServlet("/vote-meeting-time")
+public class VoteMeetingTimeServlet extends HttpServlet {
+
+  /** Increments the voteCount of the MeetingTime identified by the provided MeetingTimeId */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // expect the query string to be of the following format: ?meetingTimeId=ID&voters=VOTER
+    String keyStr = request.getParameter(MeetingTimeFields.MEETING_TIME_ID);
+    String voter = request.getParameter(MeetingTimeFields.VOTERS);
+
+    if (keyStr == null || voter == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.BAD_REQUEST_ERROR);
+      return;
+    }
+
+    Key key;
+    try {
+      key = KeyFactory.stringToKey(keyStr);
+    } catch(IllegalArgumentException e) {
+      // the keyStr is an invalid key
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.INVALID_KEY_ERROR);
+      return;
+    }
+
+    // filter by Key
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity meetingTime;
+
+    try {
+      meetingTime = datastore.get(key);
+    } catch (EntityNotFoundException e) {
+      // entity by the given key is not found
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_NOT_FOUND, ErrorMessages.ENTITY_NOT_FOUND_ERROR);
+      return;
+    }
+
+    HashSet<String> voters;
+    if (meetingTime.getProperty(MeetingTimeFields.VOTERS) == null) {
+      voters = new HashSet<String>();
+    } else {
+      voters = new HashSet<String>( (ArrayList<String>) meetingTime.getProperty(MeetingTimeFields.VOTERS));
+    }
+
+    // If this voter has voted for this time before, they cannot vote again
+    if (!voters.add(voter)) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_CONFLICT, ErrorMessages.USER_HAS_VOTED_ERROR);
+      return;
+    }
+
+    Long currentVotes = (long) meetingTime.getProperty(MeetingTimeFields.VOTE_COUNT);
```

Comment:
I got this comment on my PR: Why is it a long instead of an int?

Reply:
Also, just note that once you change it to an int, there's some parsing you need to do if you get the result back from datastore as it will be returned as a long.

Reply:
Yep, I saw what you @vivianho18 did to convert between the long fetched from Datastore to an int, so I've now done the same. It makes more sense for it to be an int, since voteCount should be a whole number of relatively small value.